core/src/main/java/org/jboss/jca/core/connectionmanager/pool/mcp/SemaphoreConcurrentLinkedQueueManagedConnectionPool.java

```diff
@@ -1126,6 +1129,35 @@ private void removeConnectionListenerFromPool(ConnectionListenerWrapper clw)
    }
 
    /**
+    * {@inheritDoc}
+    */
+   public void connectionListenerDestroyed(ConnectionListener cl)
+   {
+      if (cl.getState() == ConnectionState.DESTROYED)
+      {
+         log.tracef("ManagedConnection is already destroyed %s", cl);
+
+         return;
+      }
+
+      if (statistics.isEnabled())
```

Comment:
The same here. Only this statistics update should remain inside connectionListenerDestroyed. The remainder of the code is already in ConnectionListener.destroy